### DISCUSSION TITLE
Will we ever see new protocol that Xray team implement in their app on V2raya or NOT

### DISCUSSION THREAD

**Discussion by arminmacx** [v2rayA/v2rayA#1568] [Q&A]
Hi,

First of all thanks for this great app, I appreciated what have you done so far.

But from long time ago as i already ask the team here if we see new protocol that implemented in Xray-core in V2raya or not i dont see any  response neither any updtae to add XHTTP,SplitHTTP and other protocol into V2raya, So I ask it again if we ever gonna see any update from V2raya team regard of this or not as some of these new protocols can help  some people to bypass restriction and old one may or may not work in some country.

So we need to update the config, And since when ever we change the config of V2raya it reset it and just work with its own config we have problems which I asking the team please add these new features and in extra let users change the config at least so they can have their own protocol and only keep the Tproxy and iptables intact

**Comment by mirdukkkkk** [comment]
I've been using v2rayA for a decent amount of time now and would love to see XHTTP (SplitHTTP) in there. In addition to the suggestion, I can attach a post from RPRX (Xray-core maintainer) on how XHTTP works, and its advantages over gRPC - https://github.com/XTLS/Xray-core/discussions/4113.

Also, when running the latest version of Xray-core, you may see a message labeled ERROR of the following kind:
```
2024/12/08 23:27:57 [Info] common/errors: The feature gRPC transport will be migrated to XHTTP transport in the future.
```

So, I think the new protocol will not be superfluous in v2rayA.

**Comment by cola77777** [comment]
I'm also looking forward to seeing v2raya being able to use XHTTP

**Comment by ser-daniel** [comment]
Hi, team! What is need to be done to support new protocols? Maybe I can help

**Comment by dxcvvxd** [comment]
the way nobody gaf

**Comment by back-lacking** [comment]
bumping this
i got 2 proxies that im running 3x-ui on, and running xhttp inbounds as websocket and httpupgrade are going to be deprecated. would like to see these transports supported in the future

**Reply by aradng** [reply]
repo is abandoned by the maintainer AFAIK, so sad to see a cool product like this perish w/o alternatives

**Comment by 1andrevich** [comment]
#1727 It was added in 2.2.7, I checked in recent 2.2.7.4 version, XHTTP is working

**Reply by Anderhar** [reply]
> #1727 It was added in 2.2.7, I checked in recent 2.2.7.4 version, XHTTP is working

Yeah, but unfortunately, there is still no support for `"downloadSettings"` object to take advantage of uplink/downlink separation, which is one of the most powerful features of XHTTP. Maybe I'm wrong, but it seems v2rayA can only run XHTTP in `stream-one` mode right now, which is pretty limited.

**Comment by Mioingithub** [comment]
use the xray core. what is the problem? this is possible in v2raya. I'm just using it.

**Comment by Anderhar** [comment]
Has anyone figured out how to use the **“Extra Raw JSON”** field to add Xmux params? Seems like someone already posted the exact recipe somewhere here, but looks like it's vanished now. 

My target server is configured as XHTTP `stream-up`, but anything else added to the **“Extra Raw JSON”** field isn’t saved and doesn’t reflect in `/etc/v2raya/config.json` as well. The following variants have been tested:

```
"xmux":{...}

{"xmux":{...}}

extra={"xmux":{...}}

"xhttpSettings": {"xmux": {...}}
```

Especially would like to know the @YLinXin 's opinion who implemented this.

**Reply by YLinXin** [reply]
Maybe you can referrer this. [上下行分离](https://github.com/XTLS/Xray-core/discussions/4113)

**Reply by Anderhar** [reply]
Thank you for reaching out.
 
Of course I did, and already been using the Extra object in other clients for a while now (something like `{"xPaddingBytes":"100-1000","xmux":{"maxConcurrency":"16-32","maxConnections":"0","cMaxReuseTimes":"0","hMaxRequestTimes":"600-900","hMaxReusableSecs":"1800-3000","hKeepAlivePeriod":"0"}}` ). 

Unfortunately, V2RayA is the only one that's being stubborn about it. The point is that the text cannot even be saved, regardless of the syntax, and without any error messages as well.

**Reply by YLinXin** [reply]
Yes I have checked this repo and my repo. There is no corresponding code to handle this. I have added raw JSON object about extra message at my branch [xhttp-at-xray](https://github.com/YLinXin/v2rayA/tree/xhttp-at-xray)

I have fix your message with correct data type and do manual testing.
```
{"xPaddingBytes":"100-1000","xmux":{"maxConcurrency":"16-32","maxConnections":0,"cMaxReuseTimes":0,"hMaxRequestTimes":"600-900","hMaxReusableSecs":"1800-3000","hKeepAlivePeriod":0}}
```

Furthermore, I am sorry to all recently code is generated by AI. I just little and basic test by hand. And I have done so many changes by 3 PR, but the author have no receive them. So there are so many changes need to rebase and merge I have done so many times, but I give up the latest.

**Reply by YLinXin** [reply]
And I have no idea about these extra setting what will happened by them. So there is just correct handling and saving it but no functionality verify.

**Reply by Anderhar** [reply]
It’s reassuring to know that I’m maybe not a complete dumb (except for that dumby part with the quotations this time). It’s such a shame that I use V2RayA on OpenWRT (the best xray client on that platform), where updates always arrive last. But thank you very much for clarifying things – I’ve been missing that clarity this whole month.

**Reply by YLinXin** [reply]
I have let AI be my 'human' partner recently, and it is so helpful to handle the 'computer' thing as 'being' human.
So you can have a try if some computer problems happening.  ^_^
